Calculator/ViewController.swift

```diff
@@ -11,102 +11,93 @@ import UIKit
 class ViewController: UIViewController
 {
     @IBOutlet weak var display: UILabel!
+    @IBOutlet weak var history: UILabel!
     
     var userIsInTheMiddleOfTypingANumber = false
     var floatingPointInNumber = false
+    var brain = CalculatorBrain()
     
     @IBAction func appendDigit(sender: UIButton) {
-        let digit = sender.currentTitle!
+        if let digit = sender.currentTitle {
         if userIsInTheMiddleOfTypingANumber {
             display.text = display.text! + digit
+            history.text = history.text! + digit
         } else {
             display.text = digit
+            history.text = history.text! + digit
             userIsInTheMiddleOfTypingANumber = true
         }
+        }
     }
     @IBAction func appendFloatingPoint(sender: UIButton) {
```

Comment:
http://stackoverflow.com/questions/24034043/how-do-i-check-if-a-string-contains-another-string-in-swift